submariner/globalnet-enhancement2-0.md

```diff
@@ -0,0 +1,299 @@
+    // in case it is not set, it results in all the Pods selected from the namespace in which the
+    // CR is created.
+    // +optional
+    PodSelector metav1.LabelSelector `json:"podSelector,omitempty"`
+}
+
+const (
+    // GlobalEgressIPSuccess means that Globalnet was able to successfully allocate the GlobalIP.
+    GlobalEgressIPSuccess GlobalEgressIPStatus = "Success"
+
+    // GlobalEgressIPError means that Globalnet was unable to allocate the GlobalIP.
+    GlobalEgressIPError GlobalEgressIPStatus = "Error"
+)
+
+type EgressIPStatus struct {
+    // Status is one of {"Success", "Error"}
+    Status GlobalEgressIPStatus `json:"status,omitempty"`
+
+    // +optional
+    Message *string `json:"message,omitempty"`
+
+    // The list of GlobalIPs assigned via this GlobalnetEgressIP object.
+    GlobalIPs []string `json:"globalIPs"`
+}
+```
+
+When Submariner Globalnet is deployed in a Cluster, it auto-allocates the configured number of
+`globalIPs` and programs the necessary egress rules on the active Gateway node of the Cluster.
+This is done internally by creating `ClusterGlobalEgressIP` CR object with the name `cluster-default`.
+Users can query the CR to check the allocated GlobalIPs at the Cluster level. If admin creates
+additional ClusterGlobalEgressIP CRs that match the Cluster scope, the status field of those CRs will
+be marked as error with appropriate message. Basically only `cluster-default` object will be supported.
+
+Only an admin user can create the `ClusterGlobalEgressIP` object and regular users will only be able
+to create `GlobalEgressIP` objects.
+
+### Backward Compatibility
+
+In order to support the newer implementation, a combination of ipSets with multiple IPTable chains
+will be used. Also, we will internally use configMaps to maintain the state of allocated GlobalIPs
+and get rid of annotations on the objects. Because of this, backward compatibility cannot be maintained
+with the existing implementation of Globalnet.
+
+### User Stories
+
+1. User has two (or more) clusters with Overlapping CIDRs and requires the Pods in one
+   Cluster be able to discover and talk to exported Services.
+
+   To achieve this use-case, user can simply deploy Submariner Globalnet on all the participating
+   clusters. By default, a globalIP (or a set of globalIPs) will be assigned at the Cluster level
+   which will be used as egressIPs for inter-cluster communication and necessary Services can be
+   exported.
+
+2. In a Globalnet deployment user wants all the Pods in a namespace `ns1` to use a unique globalIP
+   instead of the globalIP assigned at the cluster level.
+
+   To achieve this, one can apply the following CRD in the namespace `ns1`.
+
+```yaml
+   apiVersion: submariner.io/v1alpha1
+   kind: GlobalEgressIP
+   metadata:
+     name: ns-egressip
+     namespace: ns1
+   spec:
+     numGlobalIPs: 1
+```
+
+3. In a Globalnet deployment, user wants a selected Pod (or set of Pods) in a particular namespace
```

Comment:
@mkolesnik, if I give an additional space between "3. In", there is a linter issue

```
submariner/globalnet-enhancement2-0.md: 213: MD029/ol-prefix Ordered list item prefix [Expected: 1; Actual: 3; Style: 1/1/1]
```

Reply:
It seems to be because you have a blank line above